File: backend/app/schemas/color.py
from pydantic import BaseModel, ConfigDict, Field
from typing import Optional
from datetime import datetime


class ColorEntryCreate(BaseModel):
    code: str = Field(
        ...,
        min_length=1,
        max_length=10,
        description="e.g., H2, F5, G7",
    )
    color_hex: str = Field(..., pattern=r"^#[0-9A-Fa-f]{6}$")
    color_name: Optional[str] = None
    sort_order: int = Field(default=0, ge=0)


class ColorEntryUpdate(BaseModel):
    code: Optional[str] = None
    color_hex: Optional[str] = None
    color_name: Optional[str] = None
    sort_order: Optional[int] = None


class ColorEntryResponse(BaseModel):
    id: int
    library_id: int
    code: str
    color_hex: str
    color_name: Optional[str] = None
    sort_order: int

    model_config = ConfigDict(from_attributes=True)


class ColorLibraryResponse(BaseModel):
    id: int
    name: str
    is_default: bool
    created_at: datetime
    entries: list[ColorEntryResponse] = []

    model_config = ConfigDict(from_attributes=True)


class ColorLibrarySummary(BaseModel):
    """Lightweight response without entries list"""

    id: int
    name: str
    is_default: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/color.ts
export interface ColorEntry {
  id: number;
  library_id: number;
  code: string;
  color_hex: string;
  color_name: string | null;
  sort_order: number;
}

export interface ColorLibrary {
  id: number;
  name: string;
  is_default: boolean;
  created_at: string;
  entries: ColorEntry[];
}

export interface ColorLibrarySummary {
  id: number;
  name: string;
  is_default: boolean;
  created_at: string;
}

export interface ColorEntryCreate {
  code: string;
  color_hex: string;
  color_name?: string | null;
  sort_order?: number;
}

export interface ColorEntryUpdate {
  code?: string;
  color_hex?: string;
  color_name?: string | null;
  sort_order?: number;
}
